lib/workers/repository/update/branch/index.ts

```diff
@@ -97,6 +122,12 @@ export async function processBranch(
     }
   }
 
+  const cache = getCache();
+  const { branches = [] } = cache;
+  const branchCache =
+    branches?.find((branch) => branch.branchName === config.branchName) ??
+    ({} as BranchCache);
+
```

Comment:
These lines represent a bit of change in design, considering that the branch cache wasn't accessed from within this function before. Consider if this logic should be moved out, or if more branch cache logic should be moved in (if more elegant to do so).

Reply:
After this PR I will create 2 if-blocks instead if one. One will check all fields related to branch upgrade another will take care of PR related changes ie. will need 2 diff functions for this. So I think better to have the branch cache logic inside processBranch so that we can pass the branchCache easily to the new functions.

Reply:
Or we can move the branchCache logic to top scope .

Reply:
this logic should be moved to a seperate file.

Shouldn't we call this in parent function? so we don't need to change so many LOC ?

eg somewhere here:

https://github.com/renovatebot/renovate/blob/0b067c6814a052a9263d0398d9fbe65acbee6f13/lib/workers/repository/process/write.ts#L38